binary_heap.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+
+
+class BinaryHeap(object):
+    """Implements a binary max-heap data structure"""
+
+    def __init__(self, an_iter=None):
+        '''Initializes the heap,
+        optionally populates it with the given iterable
+        '''
+
+        self._list = []
+
+        if an_iter:
+            for val in an_iter:
+                self.push(val)
+
+    def push(self, val):
+        '''Pushes the value into the heap,
+         adjusting the ordering to maintain the heap
+         '''
+
+        self._list.append(val)
+        child_pos = len(self._list) - 1
+
+        while child_pos and (
```

Comment:
It's traditional (and arguably better) to break the algorithm of this loop into a "helper function" that takes care of fixing ordering after the new item is appended.